dotenv.cabal

```diff
@@ -62,7 +63,7 @@ flag dev
 
 executable dotenv
   main-is:             Main.hs
-  build-depends:         base >= 4.9 && < 5.0
+  build-depends:         base >= 4.14 && < 5.0
```

Comment:
@CristhianMotoche this change appears to be effectively dropping support for GHC versions older than 8.10, I believe we should document this in the CHANGELOG.

<img width="230" alt="image" src="https://github.com/stackbuilders/dotenv-hs/assets/2049686/fb80bd10-b439-4f31-a2ea-e70608da0100">